omod/src/test/java/org/openmrs/module/idgen/web/controller/IdentifierSourceRestControllerTest.java

```diff
@@ -0,0 +1,222 @@
+		identifierSource.add("description", "test sequential identifier");
+		identifierSource.add("identifierType", IDENTIFIER_TYPE_UUID);
+		identifierSource.add("firstIdentifierBase", "1");
+		identifierSource.add("prefix", "ABC");
+		identifierSource.add("suffix", "XYZ");
+		identifierSource.add("minLength", "36");
+		identifierSource.add("maxLength", "38");
+		identifierSource.add("baseCharacterSet", "ABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789");
+		identifierSource.add("sourceType", "SequentialIdentifierGenerator");
+		
+		String jsonIdentifierSource = new ObjectMapper().writeValueAsString(identifierSource);
+				
+		MockHttpServletRequest req = newPostRequest(getURI(), jsonIdentifierSource);
+				
+		handle(req);
+				
+		assertEquals(service.getAllIdentifierSources(false).get(7).getName(), "test sequential identifier");
+		assertEquals(initialIdentifierSourceCount + 1, getAllCount());
+		
+	}
+	
+	@Test
+	public void shouldSaveARemoteIdentifierGenerator() throws Exception {
+		
+		long initialIdentifierSourceCount = getAllCount();
+		
+		SimpleObject identifierSource = new SimpleObject();
+		identifierSource.add("name", "test remote identifier");
+		identifierSource.add("description", "test remote identifier");
+		identifierSource.add("url", "http://localhost");
+		identifierSource.add("username", "openmrsuser");
+		identifierSource.add("password", "openmrspassword");
+		identifierSource.add("identifierType", IDENTIFIER_TYPE_UUID);
+		identifierSource.add("sourceType", "RemoteIdentifierSource");
+		
+		String jsonIdentifierSource = new ObjectMapper().writeValueAsString(identifierSource);
+				
+		MockHttpServletRequest req = newPostRequest(getURI(), jsonIdentifierSource);
+				
+		handle(req);
+				
+		assertEquals(service.getAllIdentifierSources(false).get(7).getName(), "test remote identifier");
+		assertEquals(initialIdentifierSourceCount + 1, getAllCount());
+		
+	}
+	
+	@Test
+	public void shouldSaveAnIdentifierPool() throws Exception {
+		
+		long initialIdentifierSourceCount = getAllCount();
+		
+		SimpleObject identifierSource = new SimpleObject();
+		identifierSource.add("name", "test identifier pool");
+		identifierSource.add("description", "test identifier pool");
+		identifierSource.add("batchSize", "500");
+		identifierSource.add("minPoolSize", "5");
+		identifierSource.add("sequential", "true");
+		identifierSource.add("refillWithScheduledTask", "true");
+		identifierSource.add("sourceUuid", getUuid());
+		identifierSource.add("identifierType", IDENTIFIER_TYPE_UUID);
+		identifierSource.add("sourceType", "IdentifierPool");
+		
+		String jsonIdentifierSource = new ObjectMapper().writeValueAsString(identifierSource);
+				
+		MockHttpServletRequest req = newPostRequest(getURI(), jsonIdentifierSource);
+				
+		handle(req);
+				
+		assertEquals(service.getAllIdentifierSources(false).get(7).getName(), "test identifier pool");
+		assertEquals(initialIdentifierSourceCount + 1, getAllCount());
```

Comment:
Can you assert that each of the properties you posted was saved?

Reply:
asserted